Cnab refactor

mileo opened the pull request.

mileo committed:
[NEW] Work in progress - Boleto registrado

fernandomr committed:
Ajustes para Itau, Cef, Bradesco -- 240
Ajustes para Bradesco Pag For 500
Sequencia de arquivos emitidos implementada via ir.sequence.

mileo committed:
Merge pull request #10 from kmee/feature/implementacao_cnab

[NEW] Work in progress - Boleto registrado

crsilveira committed:
arrumado campo valor

lepri committed:
ajustes na formatação do arquivo

lepri committed:
correção ortográfica

crsilveira committed:
adicionado novos campos para o Cnab

lepri committed:
pep8

lepri committed:
implementacao do cnab240 do santander

lepri committed:
ajuste no tamanho do endereço do sacado

crsilveira committed:
Adicionado banco Sicredi

mileo committed:
Merge pull request #15 from lepri/feature/implementacao_cnab

Feature/implementacao cnab

mileo committed:
[REF] PEP8

mileo committed:
PEP8

mileo committed:
Merge pull request #17 from odoo-brazil/sicred

Adicionado banco Sicredi

mileo committed:
[REM] Removing company logo

fernandomr committed:
sequencia de arquivo

mileo committed:
Merge branch 'teste_pag_for_bradesco500' of https://github.com/fernandomr/odoo-brazil-banking into fernandomr-teste_pag_for_bradesco500

Conflicts:
	__unported__/account_statement_l10n_br_cnab240_import/__openerp__.py
	l10n_br_account_banking_payment_cnab/__init__.py
	l10n_br_account_banking_payment_cnab/__openerp__.py
	l10n_br_account_banking_payment_cnab/febraban/cnab_240/bancos/bradesco.py
	l10n_br_account_banking_payment_cnab/febraban/cnab_240/cnab_240.py

mileo committed:
Merge branch 'fernandomr-teste_pag_for_bradesco500' into cnab

mileo committed:
[FIX] pep8 and merge clear

mileo committed:
[NEW] OCA Tests enable

[FIX] pep8

[FIX] pep8

[FIX] pep8

mileo committed:
[FIX] OCA dependencies

[FIX] OCA Dependencies

[NEW] Configure locales and fix cnab dependencies

[FIX] File name

[NEW] PyBoleto add to dependencies

[FIX]

mileo committed:
[FIX] Pyboleto

mileo committed:
Merge pull request #23 from odoo-brazil/oca-tests

[NEW] OCA Tests enable

fernandomr committed:
renomeacao de campos e adicao de informacoes

fernandomr committed:
mudanca da origem de algumas variaveis

fernandomr committed:
FIX lote reinicia a cada segmentos P e Q.

fernandomr committed:
soma dos valores dos titulos

fernandomr committed:
alteracoes no payment_mode devido à natureza do cnab

fernandomr committed:
bkp

fernandomr committed:
dígito verificador agencia/conta

fernandomr committed:
ajuste no domain para selecionar apenas faturas

fernandomr committed:
mostrar o tipo de pagamento no modo de exportaçao

fernandomr committed:
testar com configuracoes corretas

fernandomr committed:
Adicionados modos de importacao

fernandomr committed:
alterada o unique_import_id devido à erro na soma de str e int

fernandomr committed:
Merge pull request #26 from odoo-brazil/pag_for

Ajuste no domain para busca de fatura

fernandomr committed:
Merge pull request #27 from odoo-brazil/feature/impt_arq

Feature/impt arq

fernandomr committed:
ajustes pag for 500

fernandomr committed:
flake8

fernandomr committed:
Update __openerp__.py

mileo committed:
[NEW] Exportacao de pagamento de tributos

fernandomr committed:
nomeação do arquivo

fernandomr committed:
Merge branch 'cnab' of https://github.com/odoo-brazil/odoo-brazil-banking into cnab

fernandomr committed:
modificacao domain

lfdivino committed:
[ADD] Guia de recolhimento de tributo

fernandomr committed:
arrumar campos com as informações do cnab 400

lfdivino committed:
[ADD] Campos para demostar o icms_st de cada line da invoice e o total da nota fiscal

fernandomr committed:
bkp

fernandomr committed:
sequencial dos registros pagfor

fernandomr committed:
domain ajustado

fernandomr committed:
ajuste

fernandomr committed:
ajustes

fernandomr committed:
ajustar taxa de juros

mileo committed:
[WIP] Work in progress

fernandomr committed:
ajuste

fernandomr committed:
Merge pull request #2 from kmee/feature/odoo_brazil_cnab400

Feature/odoo brazil cnab400

mileo committed:
[REF] Some refactors + pep8

fernandomr committed:
fix model name

fernandomr committed:
fix

fernandomr committed:
relacao sequencia e codigo comunicacao

mileo committed:
[REF] Removing view and model [ADD] New features in pag for

fernandomr committed:
fix numero convenio cliente

fernandomr committed:
alteracoes layout apos validacao

fernandomr committed:
ajustes

mileo committed:
[NEW] WIP

mileo committed:
Merge branch 'cnab' of https://github.com/kmee/odoo-brazil-banking into cnab

mileo committed:
[WIP] add tipos de pagamentos

fernandomr committed:
impl ted

fernandomr committed:
alteracoes

fernandomr committed:
WIP alteracoes de visao e adicao de taxa de juros por linha

fernandomr committed:
Merge branch 'feature/cnab_refactory' of https://github.com/kmee/odoo-brazil-banking into feature/cnab_refactory

fernandomr committed:
dados de juros, WIP importacao 400

fernandomr committed:
valor juros, numero_documento, nome_arquivo

fernandomr committed:
ajustes do merge, domain e calculo de taxa de mora

fernandomr committed:
sequencia do arquivo e sufixo do arquivo atrelado ao modo de pagamento. numero_documento puxa o name da move_line em vez de apenas numero da fatura

fernandomr committed:
WIP pag for: modalidades 1 e 8

fernandomr committed:
wip codigos transf cc e ted

fernandomr committed:
NEW: search for all posted moves

fernandomr committed:
WIP titulos terceiros

fernandomr committed:
WIP: não busca mais move_lines já exportadas

mileo committed:
Merge branch 'cnab' into feature/pag_trib_bradesco

mileo committed:
[NEW] GNRE imported to payment.order

mileo committed:
[NEW] Add GNRE

mileo committed:
Merge pull request #3 from kmee/feature/pag_trib_bradesco

Feature/pag trib bradesco

mileo committed:
[NEW] Related field

mileo committed:
[FIX] Removing puntuation

mileo committed:
Merge pull request #4 from kmee/feature/pag_trib_bradesco

Feature/pag trib bradesco

mileo committed:
[REM] Store true

fernandomr committed:
only allow invoices already sent to sefaz

fernandomr committed:
nova view para acompanhar as linhas de cobranca

fernandomr committed:
checkbox para permitir exportar linha novamente

fernandomr committed:
WIP importacao do bradesco 400

fernandomr committed:
WIP

fernandomr committed:
WIP

fernandomr committed:
wip: importacao

fernandomr committed:
Merge pull request #5 from kmee/feature/importacao_bradesco_400

Feature/importacao bradesco 400

fernandomr committed:
novas visoes

fernandomr committed:
ajuste

fernandomr committed:
WIP enderecos de cobranca

fernandomr committed:
[NEW]: trazer endereco de cobrança do parceiro se houver um

fernandomr committed:
[NEW] Permite apenas faturas que geram financeiro

fernandomr committed:
anotacao para evitar singleton

fernandomr committed:
permite importar titulos que nao existem e mantê-los em aberto

lfdivino committed:
Refactor do codigo do cnab cobranca 400 para funcionar junto com o core e com os demais layouts do cnab.

lfdivino committed:
Refactor nos metodos para adicionar todas os retornos do cnab cobranca na tabela l10n_br_cnab.move e adicionado verificacao onde so sera criado o evento no banking_statement se o mesmo for criado no l10n_br_cnab.move

lfdivino committed:
Envio da view do l10n_br_cnab.move

lfdivino committed:
Envio do account_bank_statement_impor.py

lfdivino committed:
[FIX] Acertado a virgula nos valores dos retornos do cnab e tambem alterado para 'nosso_numero' ao inves de 'meu_numero'

lfdivino committed:
[FIX] Mudancas no cadastro de novos cnab moves e associacao entre o cnab move e o account move line

lfdivino committed:
[ADD] Adicionado arquivos da relacao do account move line com o cnab move

lfdivino committed:
[FIX] associacao entre account_move_line e l10n_br_cnab.move

lfdivino committed:
[FIX] Sequencia dos boletos do titulo na view do cnab_move

lfdivino committed:
[FIX] pequena correcao na busca das contas bancarias

lfdivino committed:
[FIX] mapeamento dos titulos para a reconciliacao

lfdivino committed:
[FIX] Alterado para o numero da carteira na busca pela conta bancaria

mileo committed:
[REF] Add commit transaction to create cnab_move

lfdivino committed:
[FIX] transaction agora recebe o valor pago pelo boleto.

lfdivino committed:
Merge branch 'cnab_refactor' of https://github.com/kmee/odoo-brazil-banking into cnab_refactor

mileo commented:
@leorochael It would be very important to you can review. Thanks in advance

danimaribeiro commented on l10n_br_account_banking_payment_bradesco_tributos/model/payment_mode.py:
```diff
@@ -0,0 +1,63 @@
+
+    payment_order_type = fields.Selection(
+        selection_add=[
+            ('tax', u'Tributos'),
```
This field is related:
https://github.com/OCA/bank-payment/blob/8.0/account_banking_payment_export/models/payment_mode.py#L73

It gives an error when trying to add new selection, should be removed.

danimaribeiro commented on l10n_br_account_payment_mode/models/payment_mode.py:
```diff
@@ -24,16 +24,35 @@
 
+    payment_order_type = fields.Selection(
+        selection_add=[
+            ('cobranca', u'Cobrança'),
```
Same here: related field.

danimaribeiro commented on l10n_br_cnab400_import/__openerp__.py:
```diff
@@ -0,0 +1,52 @@
+    'depends': [
+        'account_bank_statement_import',
+        'l10n_br_account',
+        'account_bank_statement_import_bradesco_cnab',
```
Should be removed this dependency?

danimaribeiro commented on l10n_br_account_banking_payment_cnab/__openerp__.py:
```diff
@@ -0,0 +1,55 @@
+        #'data/l10n_br_payment_export_type.xml',
+        #'data/l10n_br_payment_mode.xml',
+    ],
+    'active': False,
```
'active': False ?

danimaribeiro commented on l10n_br_account_payment_mode/__openerp__.py:
```diff
@@ -33,15 +33,14 @@
         'demo/payment_demo.xml'
-        ],
+    ],
     'active': False,
```
True?

danimaribeiro commented on l10n_br_cnab400_import/__openerp__.py:
```diff
@@ -0,0 +1,52 @@
+        'view/account_move_line_view.xml',
+        # 'data/l10n_br_res_partner_bank_type.xml',
+    ],
+    'active': False,
```
True?

danimaribeiro commented on l10n_br_cnab_import/__openerp__.py:
```diff
@@ -0,0 +1,49 @@
+    'data': [
+        'view/cnab_import_view.xml',
+    ],
+    'active': False,
```
True

danimaribeiro commented on l10n_br_account_banking_payment_cnab/wizard/payment_order_create.py:
```diff
@@ -0,0 +1,137 @@
+                ]
+            # TODO: Refactory this
+            index = domain.index(('invoice.payment_mode_id', '=', False))
+            del domain[index - 1]
```
File "/home/danimar/Projetos/odoo/kmee-brazil-banking/l10n_br_account_banking_payment_cnab/wizard/payment_order_create.py", line 42, in extend_payment_order_domain
    index = domain.index(('invoice.payment_mode_id', '=', False))
ValueError: ('invoice.payment_mode_id', '=', False) is not in list

danimaribeiro commented:
Se desmarcar a opção "Titulos Abertos" na tela "Preencher Pagamento" ocorrre o seguinte erro:

```
 AssertionError: This domain is syntactically not correct: 
 [('move_id.state', '=', 'posted'), ('reconcile_id', '=', False), 
 ('company_id', '=', 1), '|', ('debit', '>', 0)]
```

To só escrevendo para registrar aqui os problemas, conforme vou corrigindo vou enviando PR.

danimaribeiro commented on l10n_br_account_banking_payment_cnab/wizard/payment_order_create.py:
```diff
@@ -0,0 +1,137 @@
+            index = domain.index(('invoice.payment_mode_id', '=', False))
+            del domain[index - 1]
+            domain.remove(('invoice.payment_mode_id', '=', False))
+            index = domain.index(('date_maturity', '<=', self.duedate))
```
Why remove this here? The other types are commented
I think is better to apply the filter.

danimaribeiro commented on l10n_br_account_banking_payment/wizard/payment_order_create.py:
```diff
@@ -0,0 +1,184 @@
+        # Search for all posted moves
+        if self.all_posted_moves == True:
+            index = domain.index(('date_maturity', '<=', self.duedate))
+            domain[index + 1] = ('date_maturity', '>', self.duedate)
```
"self.all_posted_moves" é o campo "Titulos abertos", porque quando ele está marcado utiliza esse dominio?
isso está restringindo a pesquisa apenas aos títulos de hoje e não aos abertos.

fernandomr replied:
Danimar, a pesquisa quando este campo está marcado deve incluir todas as move_lines em aberto. Independente da data de vencimento dele.
É um recurso para o seguinte caso: vamos supor que na última sexta não tenha sido gerada uma remessa de cobrança mas existem itens no sistema que precisam ser exportados. Essa pesquisa pelo all_posted_moves inclui esses itens na remessa de hoje, por exemplo.

lfdivino committed:
[FIX] Verificacao do codigo do evento para adicionar a transactions do bank_statement

lfdivino committed:
[ADD] Xml que cria o tipo de conta bancaria 'Retorno CNAB' e as configuracoes de fields obrigatorios desta opcao.

danimaribeiro commented:
Quando eu crio uma ordem de cobrança e adiciono os títulos eles ficam negativos, o que não permite exportar.
![captura de tela de 2016-04-10 09-32-49](https://cloud.githubusercontent.com/assets/854781/14410189/3cc90e38-feff-11e5-9e06-f404da1ea937.png)

fernandomr committed:
[REF] Alterção do retorno de remessa para busca somente por transaction_ref
- Agregando maior compatibilidade com oca/base_transaction_id
- Aumentando a segurança do retorno de forma que somente o seja reconciliado por numero bancário.

fernandomr committed:
alteracoes para pesquisa por cnab_move

fernandomr committed:
Move.lines search based only on transaction_ref value.
The move.lines search depending on amount should not occur in this case.

fernandomr committed:
Fix: transaction_ref cannot be unique on move.line. Different bank accounts might return equal transaction_ref on cnab returns

fernandomr committed:
Volta do nome do título na importação.
Busca da move_line pelo nome também. Importante pelo fato de duas contas bancárias poderem retornar o mesmo transaction_ref

yk2kus commented on l10n_br_account_payment_mode/views/payment_mode_view.xml:
```diff
@@ -1,37 +1,70 @@
-										attrs="{'invisible': [('type_payment', '!=', '00')]}" />
-									<field name="instrucoes"/>
+                                    <field name="type_sale_payment"/>
+                                    <field name="internal_sequence_id"
```
Duplicated Field "internal_sequence_id"

yk2kus commented on l10n_br_account_payment_mode/views/payment_mode_view.xml:
```diff
@@ -1,37 +1,70 @@
+                            <page string="Pagamento" attrs="{'invisible': [('purchase_ok', '!=', True)]}">
+                                <group>
+                                    <field name="type_purchase_payment"/>
+                                    <field name="internal_sequence_id"/>
```
Duplicated Field " internal_sequence_id"

mileo committed:
[FIX] Dependencies

mileo committed:
[FIX] Corrige o dominio de pesquisa do account.move.line

mileo commented on l10n_br_cnab400_import/models/account_bank_statement_import.py:
```diff
@@ -228,7 +228,10 @@ def buscar_account_move_line(self, referencia_invoice, partner=False):
         move_line_item = move_line_model.search(
-            [('name', '=', move_line_name)], limit=1)
+            [('name', '=', move_line_name),
+             ('debit', '>', 0),
```
@fernandomr  Da uma olhada nessa alteração

fernandomr committed:
FIX sinal de grau no endereco

fernandomr committed:
Merge branch 'cnab_refactor' of https://github.com/kmee/odoo-brazil-banking into cnab_refactor
